src/plots/cartesian/axes.js

```diff
@@ -362,10 +363,30 @@ axes.doAutoRange = function(ax) {
         // doAutoRange will get called on fullLayout,
         // but we want to report its results back to layout
 
-        var axIn = ax._input;
+        axIn = ax._input;
         axIn.range = ax.range.slice();
         axIn.autorange = ax.autorange;
     }
+
+    if(ax._anchorAxis && ax._anchorAxis.rangeslider) {
+        var axeRangeOpts = ax._anchorAxis.rangeslider[ax._name];
+        if(axeRangeOpts) {
+            switch(axeRangeOpts.rangemode) {
+                case 'auto':
+                    axeRangeOpts.range = hasDeps ? axes.getAutoRange(ax) : ax.range.slice();
+                    break;
+                case 'match':
+                    axeRangeOpts.range = ax.range.slice();
+                    break;
+                default:
+                    axeRangeOpts.range = ax.range.slice();
+                    break;
```

Comment:
`case 'match'` and `default` are identical, is that expected? If so this can probably be condensed to a simple `if(rangemode === 'auto')`. Regardless, we should have a jasmine test for each of the three `rangemode` cases.

Reply:
Indeed, I forgot to refactor it down to a simple if (initialy there was more case).

Reply:
What kind of jasmine test do you want for the 3 rangemode value ?